### PR TITLE
Fix workspace integration failed due to the git safe.directory error

### PR DESCRIPTION
**Pull request by jenting**
## Description
The workspace integration failed related to Git operation.
https://werft.gitpod-dev.com/job/gitpod-workspace-run-integration-tests-main.4

Recently, the Git 2.35.2 _or_ 2.36 changes the behavior on it's stricter checks the repository ownership
 https://github.blog/2022-04-18-highlights-from-git-2-36/

I think once this PR #10012 be deployed to production, we no longer need to manually configure the `safe.directory` manually in the code. 

## Related Issue(s)

Related to #10012

## How to test
```shell
./dev/preview/install-k3s-kubeconfig.sh
cd test

TOKEN=`kubectl get secret integration-test-user -n werft -ojsonpath={.data.token} | base64 --decode`

USER_TOKEN=$TOKEN go test -v ./... -run=TestGitActions -kubeconfig=/home/gitpod/.kube/config -namespace=default -username=integration-test-user
USER_TOKEN=$TOKEN go test -v ./... -run=TestGitHubContexts -kubeconfig=/home/gitpod/.kube/config -namespace=default -username=integration-test-user
```

## Release Notes

```release-note
None
```

## Documentation

N/A

### CODE REVIEW

**Comment by roboquat** [comment]
@jenting: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

File: test/pkg/integration/common/git-client.go [hunk 1]
```diff
@@ -108,7 +123,7 @@ func (g GitClient) Commit(dir string, message string, all bool) error {
 		return err
 	}
 	if resp.ExitCode != 0 {
-		return xerrors.Errorf("commit returned rc: %d", resp.ExitCode)
+		return fmt.Errorf("commit returned rc: %d err: %v", resp.ExitCode, resp.Stderr)
```
Comment (princerachit) [inline test/pkg/integration/common/git-client.go:126]: why fmt instead of xerrors? I believe xerrors is preferred?

Comment (jenting) [reply test/pkg/integration/common/git-client.go:126]: I check recently PR deprecated xerrors.

**Review by princerachit — APPROVED** [review]
Did not run the test. Code changes lgtm

**Pull request merged**